artificer/src/test/java/net/technolords/tools/artificer/reference/JavaVersionManagerTest.java

```diff
@@ -172,21 +179,82 @@ public void testMagicNumberExtractionWithValidClasses(String fileName, String ex
+
+            // Create a resource reference linking to the file
+            Resource resource = new Resource();
+            resource.setPath(pathToResourceLocation);
+
+            // Find the magic number associated to the resource
+            JavaVersionManager javaVersionManager = new JavaVersionManager(KNOWN_JAVA_VERSIONS_REFERENCE_FILE);
+            String actualVersion = javaVersionManager.getMagicNumber(resource);
 
+        } catch (Exception e) {
+            Assert.assertEquals(e.getClass(), expectedException);
+            LOGGER.debug("Exception caused by : " + e.getCause() + " and: " + e.getMessage());
+        }
+    }
+    /** Data Set to test the registered Java Versions */
+    @DataProvider (name = "dataSetWithCompiledVersions")
+    public Object[][] dataSetWithCompiledVersions() {
+        return new Object[][] {
+                { "Analysis.class", "1.8"},
+                { "RfuRouteBuilder.class", "1.7"}
+        };
+    }
     /**
      * Test case 6: Test the registration of the java versions in the model.
      */
-    @Test
-    public void testRegistrationOfjavaVersionInModel() {
-        // TODO: implement
+
+    @Test(dataProvider = "dataSetWithCompiledVersions")
+    public void testRegistrationOfJavaVersionInModel(String fileName, String expectedCompiledVersion){
+
+        Path pathToResourceLocation = FileSystems.getDefault().getPath(this.pathToDataFolder.toAbsolutePath() + File.separator + fileName);
+        LOGGER.debug("The path towards the class file '" + fileName + "' exists: " + Files.exists(pathToResourceLocation));
+
+        Resource resource = new Resource();
+        resource.setPath(pathToResourceLocation);
+
+        Meta meta = new Meta();
+
+        JavaVersionManager javaVersionManager = new JavaVersionManager(KNOWN_JAVA_VERSIONS_REFERENCE_FILE);
+        javaVersionManager.registerCompiledVersion(meta, resource);
+
+        Assert.assertEquals(resource.getCompiledVersion(), expectedCompiledVersion);
+        LOGGER.debug("Resource Compiled Version = " + resource.getCompiledVersion());
+/**
+    The below statements were a trial to retrieve the Found Version but I am unsuccessful.
+    I can see the foundVersion in the Debug mode, but could not retrieve the value.
+    I wanted to assert on meta foundVersion, but instead used resource.getCompiledVersion.
+//ToDO : need to implement assertion on found version and total number of classes
+ */
+//        FoundJavaVersion foundJavaVersion = new FoundJavaVersion();
+//
+//        LOGGER.debug("meta.getStatus() = " + meta.getStatus());
+//        LOGGER.debug("meta.getFoundJavaVersions() = " + meta.getFoundJavaVersions().getFoundJavaVersionList().size());
+//        LOGGER.debug("Index of Found JavaVersion = " + meta.getFoundJavaVersions().getFoundJavaVersionList().indexOf(foundJavaVersion));
+//        LOGGER.debug("Found versions = "+ meta.getFoundJavaVersions().getFoundJavaVersionList());
```

Comment:
You are almost there! The model (Meta.java) says:
```
List<FoundJavaVersion> foundJavaVersionList;
```
Which means it is just a list of FoundJavaVersion objects and there is no particular index. What you need to do is to inspect each of those FoundJavaVersions in the list and see if its property is equal to the one you are looking for. The model (FoundJavaVersion.java) says:
```
public class FoundJavaVersion {
    private String foundJavaVersion;
    private long totalClasses = 0;
```
So the solution is to create a loop... and inside that loop, do something like:
foundJavaVersion.getFoundJavaVersion().equals(expectedCompiledVersion)

I am expected the following assertions to be added:
- the list size of meta.getFoundJavaVersions() to be correct
- the foundJavaVersion.getTotalClasses() to be correct